openupgrade_framework/odoo_patch/odoo/modules/graph.py

```diff
@@ -1,108 +1,21 @@
-# flake8: noqa
-# pylint: skip-file
-
-import logging
+# Copyright Odoo Community Association (OCA)
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
 import odoo
-import odoo.tools as tools
-from odoo.tools.safe_eval import safe_eval
-
 from odoo.modules.graph import Graph
 
-_logger = logging.getLogger(__name__)
-
-
-if True:
-
-    def _update_from_db(self, cr):
-        if not len(self):
-            return
-        # update the graph with values from the database (if exist)
-        ## First, we set the default values for each package in graph
-        additional_data = {key: {'id': 0, 'state': 'uninstalled', 'dbdemo': False, 'installed_version': None} for key in self.keys()}
-        ## Then we get the values from the database
-        cr.execute('SELECT name, id, state, demo AS dbdemo, latest_version AS installed_version'
-                   '  FROM ir_module_module'
-                   ' WHERE name IN %s',(tuple(additional_data),)
-                   )
-
-        ## and we update the default values with values from the database
-        additional_data.update((x['name'], x) for x in cr.dictfetchall())
-
-        # <OpenUpgrade:ADD>
-        # Prevent reloading of demo data from the new version on major upgrade
-        if ('base' in self and additional_data['base']['dbdemo'] and
-                additional_data['base']['installed_version'] <
-                odoo.release.major_version):
-            cr.execute("UPDATE ir_module_module SET demo = false")
-            for data in additional_data.values():
-                data['dbdemo'] = False
-        # </OpenUpgrade>
 
+def update_from_db(self, cr):
+    """ Prevent reloading of demo data from the new version on major upgrade """
+    Graph.update_from_db._original_method(self, cr)
+    if (
+        "base" in self
+        and self["base"].dbdemo
+        and self["base"].installed_version < odoo.release.major_version
```

Comment:
🚫 Beware! You have to use a smart version comparison algorithm. Check out `odoo.tools.parse_version`.